android/src/main/java/com/swmansion/rnscreens/animator/CustomAnimatorSetProvider.kt

```diff
@@ -0,0 +1,239 @@
+package com.swmansion.rnscreens.animator
+
+import android.animation.AnimatorSet
+import android.animation.ObjectAnimator
+import android.content.Context
+import android.view.animation.AnimationUtils
+import com.swmansion.rnscreens.R
+import com.swmansion.rnscreens.Screen
+import com.swmansion.rnscreens.ext.parentAsView
+
+class CustomAnimatorSetProvider {
+    companion object {
+        fun customize(
+            context: Context?,
+            animator: Int,
+            initialAnimatorSet: AnimatorSet,
+            screen: Screen,
+        ): AnimatorSet {
+            val finalAnimatorSet = AnimatorSet()
+            val screenParent = screen.parentAsView()
+
+            // TODO: verify that these values can be used to get width/height
+            val screenWidth = screen.measuredWidth
+
+            // TODO: first value available when entering, second when exiting; last fallback
+            //       is off by header's height
+            val screenHeight = screen.container?.height ?: screenParent?.height ?: screen.measuredHeight
+
+            when (animator) {
+                R.animator.rns_default_enter_in -> {
+                    finalAnimatorSet.playTogether(
+                        initialAnimatorSet,
+                        ObjectAnimator
+                            .ofFloat(screenParent, "translationX", 0.1f * screenWidth, 0.0f)
+                            .apply {
+                                duration = 450
+                                interpolator = AnimationUtils.loadInterpolator(context, android.R.interpolator.fast_out_extra_slow_in)
+                            },
+                    )
+                }
+                R.animator.rns_default_enter_out -> {
+                    finalAnimatorSet.playTogether(
```

Comment:
Why is some places do you use `playTogether` and `play` in the others?

Reply:
okay, I see.

The way to go here, would be to keep definition of an animation in single place. Let's not split them into XML & code part.